components/crosspost-item.js

```diff
@@ -0,0 +1,65 @@
+import useCrossposter from './use-crossposter'
+import Dropdown from 'react-bootstrap/Dropdown'
+import { ITEM } from '../fragments/items'
+import { gql, useQuery, useMutation } from '@apollo/client'
+import { useToast } from './toast'
+
+export default function CrosspostDropdownItem ({ item }) {
+  const { data } = useQuery(ITEM, { variables: { id: item.id } })
+  const toaster = useToast()
+
+  const [upsertNoteId] = useMutation(
+    gql`
+      mutation upsertNoteId($id: ID!, $noteId: String!) {
+        upsertNoteId(id: $id, noteId: $noteId) {
+          id
+          noteId
+        }
+      }`
+  )
+
+  const crossposter = useCrossposter()
+
+  return !item?.noteId
+    ? (
+      <Dropdown.Item
+        onClick={async () => {
+          try {
+            if (!(await window.nostr.getPublicKey())) {
+              throw new Error('not available')
+            }
+          } catch (e) {
+            toaster.danger(`Nostr extension error: ${e.message}`)
+            return
+          }
+          try {
+            if (item?.id) {
+              const crosspostResult = await crossposter({ ...data.item })
```

Comment:
If I understand this correctly, this will post _anything_ under my own nostr pubkey as if it's my own content, right?

Shouldn't there at least be some attribution to the OP? Like a link back to the SN post?